src/main/java/com/example/raiseTechcoursetask10/controller/SkiresortController.java

```diff
@@ -52,17 +54,26 @@ public ResponseEntity<Map<String, String>> noResourceFound(
         return new ResponseEntity(body, HttpStatus.NOT_FOUND);
     }
 
-
     @PostMapping("/skiresorts")
-    public ResponseEntity<SkiresortResponse> createSkiresort(@RequestBody SkiresortCreateForm skiresortCreateForm) {
+    public ResponseEntity<SkiresortResponse> createSkiresort(@RequestBody SkiresortCreateForm skiresortCreateForm, HttpServletRequest request) {
         Skiresort skiresort = skiresortService.createSkiresort(skiresortCreateForm);
 
-        // skiresortを作成する処理
+        // skiresortオブジェクトを元にレスポンス用のオブジェクトを生成
         SkiresortResponse skiresortResponse = new SkiresortResponse(skiresort);
-        URI url = UriComponentsBuilder.fromUriString("http://localhost8080")
+        // HttpServletRequestのインスタンスでリクエストの中身を取得し、動的なURLを生成
+        URI url = UriComponentsBuilder.fromUriString(request.getRequestURI())
                 .path("/skiresorts/{id}")
                 .buildAndExpand(skiresort.getId())
                 .toUri();
         return ResponseEntity.created(url).body(skiresortResponse);
     }
+
+    @PatchMapping("/skiresorts/{id}")
+    public ResponseEntity<Map<String, String>> update(@PathVariable("id") int id, @RequestBody @Validated SkiresortCreateForm form) {
```

Comment:
更新APIなのに使っているFormの名前がSkiresortCreateFormであるのは不自然です。

更新専用のFormを作るか、
```suggestion
    public ResponseEntity<Map<String, String>> update(@PathVariable("id") int id, @RequestBody @Validated SkiresortUpdateForm form) {
```
Createと共通化するのがいいと思います。
```suggestion
    public ResponseEntity<Map<String, String>> update(@PathVariable("id") int id, @RequestBody @Validated SkiresortForm form) {
```

Reply:
変更するのを見落としていました。
修正します。